### PR TITLE
tracker(dm): respect CHARSET when build TableInfo

### LINKED ISSUE #4637: DM may report error of `[types:1074]Column length too big for column '<name>' (max = 16383); use BLOB or TEXT instead`
### What did you do?

https://asktug.com/t/topic/573546

downstream `CREATE TABLE t (c VARCHAR(20000)) CHARSET=utf8` and start task

### What did you expect to see?

no error

### What did you see instead?

[types:1074]Column length too big for column '<name>' (max = 16383); use BLOB or TEXT instead

### Versions of the cluster

DM v5.4

### current status of DM cluster (execute `query-status <task-name>` in dmctl)

```console
(paste current status of DM cluster here)
```

### PR DESCRIPTION
**Pull request by lance6716**
Signed-off-by: lance6716 <lance6716@gmail.com>

### What problem does this PR solve?

Issue Number: close #4637

### What is changed and how it works?

found that MockTableInfo will always use default charset and collation, BuildTableInfoFromAST is better

### Check List 

Tests 

 - Unit test
 - Integration test

Code changes

Side effects

Related changes

 - Need to cherry-pick to the release branch

### Release note 

```release-note
Fix a bug that long varchar will report error of "Column length too big..."
```

### CODE REVIEW

**Review by lichunzhu — APPROVED** [review]
LGTM

File: dm/pkg/schema/tracker.go [hunk 1]
```diff
@@ -484,10 +483,11 @@ func (tr *Tracker) getTableInfoByCreateStmt(tctx *tcontext.Context, tableID stri
 		return nil, dmterror.ErrSchemaTrackerInvalidCreateTableStmt.Delegate(err, createStr)
 	}
 
-	ti, err := ddl.MockTableInfo(mock.NewContext(), stmtNode.(*ast.CreateTableStmt), mockTableID)
```
Comment (WizardXiao) [inline dm/pkg/schema/tracker.go:487]: maybe can delete `mockTableID `?

**Review by D3Hunter — APPROVED** [review]

**Review by WizardXiao — APPROVED** [review]
LGTM

**Review by okJiang — APPROVED** [review]

**Pull request merged**